src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1407,11 +1407,23 @@ impl<'a, 'tcx> Visitor<'tcx> for Checker<'a, 'tcx> {
                     }
                 }
                 ty::FnPtr(_) => {
+                    let unleash_miri = self
+                        .tcx
+                        .sess
+                        .opts
+                        .debugging_opts
+                        .unleash_the_miri_inside_of_you;
```

Comment:
@oli-obk Can we make a method on tcx out of checking this flag...?

Reply:
There are a couple of places in this file where this is checked so I think cleaning this up would make sense.

Reply:
I'll make a PR for this which we can merge faster and then you can rebase and use it.

Reply:
I thought we had a nice macro for this?

Reply:
Yea we do (`unleash_miri`), and it can be used in this case, but there's still some repetition that we can avoid still.

Reply:
The repetition is removed in #64470, too

Reply:
Great 🎉